src/Datadog.Trace/Propagation/MultiplexSpanContextPropagator.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Propagation
+{
+    internal class MultiplexSpanContextPropagator : IPropagator
+    {
+        private readonly ICollection<IPropagator> _propagators;
+
+        public MultiplexSpanContextPropagator(ICollection<IPropagator> propagators)
+        {
+            _propagators = propagators;
+        }
+
+        public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> getter)
+        {
+            foreach (IPropagator propagator in _propagators)
+            {
+                SpanContext context = propagator.Extract(carrier, getter);
+
+                // if context is missing, propagator wasn't capable to extract
+                if (context != null)
+                {
+                    return context;
+                }
+            }
+
+            // non of the propagators found specific headers
+            return null;
```

Comment:
I think this is incorrect. It will return the first found context. It should aggregate them somehow:

```
            SpanContext context = null;
            foreach (IPropagator propagator in _propagators)
            {
                var extractedContext = propagator.Extract(carrier, getter);
                context = Aggregate(context, extractedContext);
            }
            return context;
```

Reply:
You cannot aggregate SpanContext. You can just pick either the first or the last one. First seems more logical and performant.

Reply:
Also see https://github.com/open-telemetry/opentelemetry-specification/issues/496